fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeManager.java

```diff
@@ -1274,20 +1293,41 @@ protected PrivilegeCollection loadPrivilegeCollection(UserIdentity userIdentity,
         PrivilegeCollection collection = new PrivilegeCollection();
         userReadLock();
         try {
-            UserPrivilegeCollection userCollection = getUserPrivilegeCollectionUnlocked(userIdentity);
-            collection.merge(userCollection);
+            UserPrivilegeCollection userPrivilegeCollection = getUserPrivilegeCollectionUnlocked(userIdentity);
+            collection.merge(userPrivilegeCollection);
             roleReadLock();
             try {
                 // 1. get all parent roles by default, but can be specified with `SET ROLE` statement
                 if (roleIds == null) {
-                    roleIds = userCollection.getAllRoles();
+                    roleIds = userPrivilegeCollection.getAllRoles();
                 }
 
                 // 2. get all predecessors base on step 1
-                roleIds = getAllPredecessorsUnlocked(roleIds);
+                Set<Long> parentRoleIds = new HashSet<>();
```

Comment:
before find all the parent role ids, we should remove the role id from `rolesIds` set which doesn't belong to this user any more?